component/src/main/java/org/ballerinalang/sap/nativeimpl/producer/RFCMetaDataParser.java

```diff
@@ -0,0 +1,273 @@
+import static org.ballerinalang.sap.utils.SapConstants.IMPORT;
+import static org.ballerinalang.sap.utils.SapConstants.STRUCTURE;
+import static org.ballerinalang.sap.utils.SapConstants.TABLE;
+import static org.ballerinalang.sap.utils.SapConstants.TABLES;
+
+/** This will contains the required methods that can be use to parser a
+ * meta data description of a BAPI/RFC call.
+ * So the BNF grammar for the meta data would looks like :
+ *  bapi Function   -> import | tables | both
+ *  import    -> structure | field | both
+ *  structure -> 1 or more fields
+ *  tables    -> 1 or more table
+ *  table     -> row
+ *  row       -> 1 or more fields
+ *  field     -> name and value
+ *
+ */
+public class RFCMetaDataParser {
+
+    private static Log log = LogFactory.getLog(RFCMetaDataParser.class);
+
+    /**
+     * process the document
+     * @param document the document node
+     * @param function the RFC function to execute
+     */
+    static void processMetaDataDocument(OMElement document, JCoFunction function) {
+
+        Iterator itr = document.getChildElements();
+        while (itr.hasNext()) {
+            OMElement childElement = (OMElement) itr.next();
+            processElement(childElement, function);
+        }
+    }
+
+    /**
+     * Process every element in the document
+     * @param element the OMElement
+     * @param function the RFC function to execute
+     */
+    private static void processElement(OMElement element, JCoFunction function) {
+
+        String qname = element.getQName().toString();
+        if (qname != null) {
+            if (qname.equals(IMPORT)) {
+                processImport(element, function);
+            } else if (qname.equals(TABLES)) {
+                processTables(element, function);
+            } else {
+                log.warn("Unknown meta data type tag :" + qname + " detected. " +
+                        "This meta data element will be discarded!");
+            }
+        }
+    }
+
+    /**
+     * Process the import function
+     * @param element the OMElement
+     * @param function the RFC function to execute
+     */
+    private static void processImport(OMElement element, JCoFunction function) {
+
+        Iterator itr = element.getChildElements();
+        while (itr.hasNext()) {
+            OMElement childElement = (OMElement) itr.next();
+            String qname = childElement.getQName().toString();
+            String name = childElement.getAttributeValue(SapConstants.NAME_Q);
+            switch (qname) {
+                case STRUCTURE:
+                    processStructure(childElement, function, name);
+                    break;
+                case FIELD:
+                    processField(childElement, function, name);
+                    break;
+                default:
+                    log.warn("Unknown meta data type tag :" + qname + " detected. " +
+                            "This meta data element will be discarded!");
+                    break;
```

Comment:
We can remove this `break` as it does not have any effect (break after default).